lib/Dialect/TritonStructured/IR/TritonStructuredOps.cpp

```diff
@@ -47,11 +53,20 @@ void LoadOp::build(OpBuilder &b, OperationState &state, Value ptr,
 
   dispatchIndexOpFoldResults(dims, dynamicDims, staticDims);
 
-  auto ptrTensorType = cast<RankedTensorType>(ptr.getType());
-  auto elemType = cast<triton::PointerType>(ptrTensorType.getElementType())
-                      .getPointeeType();
-  auto resType = RankedTensorType::get(ptrTensorType.getShape(), elemType);
+  auto ptrTensorType = dyn_cast<RankedTensorType>(ptr.getType());
+  auto tensorPtrType = dyn_cast<triton::PointerType>(ptr.getType());
```

Comment:
nit: this code is really hard to read due to the very similar naming, can we add a comment for each condition indicating whether we're dealing with blockptr or normal ptr? 😄